lib/truffle/truffle/patching.rb

```diff
@@ -0,0 +1,38 @@
+module Truffle::Patching
+  TRUFFLE_PATCHES_DIRECTORY = File.join(Truffle::Boot.ruby_home, 'lib', 'patches')
+  TRUFFLE_PATCHES           = Dir.glob(File.join(TRUFFLE_PATCHES_DIRECTORY, '**', '*.rb')).
+      select { |path| File.file? path }.
+      map { |path| path[(TRUFFLE_PATCHES_DIRECTORY.size+1)..-4] }.
+      reduce ({}) { |h, v| h.update v => true }
+end
+
+module Kernel
+
+  private
+
+  alias_method :require_without_patching, :require
```

Comment:
I think's it's worth adding some more unique prefix to this: `require_without_truffle_patching`